Helper/PaymentMethods.php

```diff
@@ -316,7 +323,7 @@ protected function getCurrentCountryCode($store, $country = null)
             return $country;
         }
 
-        if ($country = $this->getQuote()->getBillingAddress()->getCountry()) {
+        if ($isCustomerLoggedIn && $country = $this->getQuote()->getBillingAddress()->getCountry()) {
```

Comment:
Instead of using the session, can we check if customer is a guest customer using the quote?
Using the session here might cause some issues related to headless merchants

Reply:
good one, thanks!